components/ActionSheet/src/private/MDCActionSheetHeaderView.m

```diff
@@ -0,0 +1,258 @@
+  self.scrollView.contentSize = contentSize;
+
+  boundsSize.width = boundsSize.width - kLeadingPadding - kTrailingPadding;
+  CGSize titleSize = [titleLabel sizeThatFits:boundsSize];
+  titleSize.width = boundsSize.width;
+  CGSize messageSize = [messageLabel sizeThatFits:boundsSize];
+  messageSize.width = boundsSize.width;
+  boundsSize.width = boundsSize.width + kLeadingPadding + kTrailingPadding;
+
+  CGRect titleFrame = CGRectMake(kLeadingPadding, kStandardPadding,
+                                 titleSize.width, titleSize.height);
+  CGRect messageFrame = CGRectMake(kLeadingPadding, CGRectGetMaxY(titleFrame) + kMiddlePadding,
+                                   messageSize.width, messageSize.height);
+  titleLabel.frame = titleFrame;
+  messageLabel.frame = messageFrame;
+
+  CGRect scrollViewRect = CGRectZero;
+  scrollViewRect.size = self.scrollView.contentSize;
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    scrollViewRect.origin.x = scrollViewRect.origin.x + self.safeAreaInsets.left;
+  }
+#endif
+
+  self.scrollView.frame = scrollViewRect;
+  CGRect originalFrame = self.frame;
+  originalFrame.size.height = scrollViewRect.size.height;
+  self.frame = originalFrame;
+}
+
+- (CGFloat)accomodateSafeAreaInWidth:(CGFloat)width {
+  CGFloat newWidth = width;
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+  if (@available(iOS 11.0, *)) {
+    newWidth = newWidth - self.safeAreaInsets.left - self.safeAreaInsets.right;
+  }
+#endif
+  return newWidth;
+}
+
+- (void)setTitle:(NSString *)title {
+  titleLabel.text = title;
+  [self setNeedsLayout];
+}
+
+- (NSString *)title {
+  return titleLabel.text;
+}
+
+- (void)setMessage:(NSString *)message {
+  messageLabel.text = message;
+  [self setNeedsLayout];
+}
+
+- (NSString *)message {
+  return messageLabel.text;
+}
+
+- (void)setTitleFont:(UIFont *)titleFont {
+  _titleFont = titleFont;
+  [self updateTitleFont];
+}
+
+- (void)setMessageFont:(UIFont *)messageFont {
+  _messageFont = messageFont;
+  [self updateMessageFont];
+}
+
++ (UIFont *)titleFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleSubheadline];
+  }
+  return [MDCTypography subheadFont];
+}
+
++ (UIFont *)messageFontDefault {
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:MDCFontTextStyleBody1];
+  }
+  return [MDCTypography body1Font];
+}
+
+- (void)updateFonts {
+  [self updateTitleFont];
+  [self updateMessageFont];
+}
+
+- (void)updateTitleFont {
+  UIFont *titleFont = _titleFont ?: [[self class] titleFontDefault];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    titleLabel.font =
+    [titleFont mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleSubheadline
```

Comment:
nit: indent +4 here and in updateMessageFont

Reply:
Done